### PR TITLE
Support management port for health and metrics

### LINKED ISSUE #19334: Support management port for health and metrics in Quarkus 3
### Description

In the Quarkus distribution the management and health endpoints are served from the standard HTTP endpoint. 

While security best practices require those endpoints not to be accessible from the outside, Keycloak still exposes the HTTP endpoint on OpenShift as a passthrough endpoint, which makes them accessible from the outside, which requires users to either disable those endpoints, use another loadbalancer in front of OpenShift, or configure a custom route. 

Quarkus 3.0.0.Beta1 finally supports a management port which allows service health and metrics from a different port, which is then not accessible from the outside: https://github.com/quarkusio/quarkus/issues/13602

### Discussion

_No response_

### Motivation

Security best practices and hardening.

### Details

The pull request https://github.com/quarkusio/quarkus/pull/30506 adds the necessary documentation for this.

The Operator needs to pass additional options to Keycloak to make this work, and Keycloak possibly needs to support additional parameters / a different internal handling to make this work.
```[tasklist]
### Follow-up tasks
- [ ] https://github.com/keycloak/keycloak/pull/28213
- [ ] https://github.com/keycloak/keycloak/issues/28404
```

### PR DESCRIPTION
**Pull request by mabartos**
Closes #19334

## Summary

- Management interface is turned **ON** by default
(https://github.com/keycloak/keycloak/pull/27629/files#r1538915621))
- Added new  named management port to k8s Service - not exposed by Ingress 
- Operator first-class citizens only: `port`
- When TLS is configured for the main server, the management interface uses HTTPS
- When TLS properties for the management interface are not explicitly set, they're **inherited** from the **main HTTP server**
- When accessing the root of the management interface, the text `Keycloak Management Interface` is shown
- Docs will be done in a separate PR

### CODE REVIEW

**Comment by mabartos** [comment]
@keycloak/cloud-native @ahus1 JFYI - the majority of the work is done. But it needs to be polished. It'd be great to review at least the semantics around this. Thanks!

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ManagementPropertyMappers.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                        .build(),
+                // HTTPS
+                fromOption(ManagementOptions.MANAGEMENT_HTTPS_ENABLED)
+                        .isEnabled(ManagementPropertyMappers::isManagementEnabled, MANAGEMENT_ENABLED_MSG)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .transformer(ManagementPropertyMappers::getHttpsEnabledTransformer)
+                        .build(),
+                fromOption(ManagementOptions.MANAGEMENT_HTTPS_CLIENT_AUTH)
+                        .isEnabled(ManagementPropertyMappers::isHttpsEnabled, MANAGEMENT_HTTPS_ENABLED_MSG)
+                        .mapFrom("quarkus.http.ssl.client-auth")
```
Comment (mabartos) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ManagementPropertyMappers.java:68]: We should use the KC props as their transformation is made the same way.

**Comment by ahus1** [comment]
@mabartos - I'm happy to see this feature evolve. It's been a while since I've added it to my wish-list. 

I think enabling TLS for the management port is useful, so exposing for adding this. This includes also the client authentication. 

The option I think is least useful is `management-relative-path`. Still people might put it behind a reverse proxy, and then they might want to have it, so I'd be ok to keep it.

Something that surprised me was that the management port answers with a `404`, which makes it difficult to test for an admin if it has been set up correctly. I suggest to have a minimal `200` response, maybe something like this: 
a9d19ae4bf5cc74a7fe49efcf3379c7a60f11516 - this is an adapted snippet from the Quarkus docs, see https://quarkus.io/guides/management-interface-reference

Thinking about it a bit more, it should answer to `management-relative-path` configurable path, and the root path should redirect to that `management-relative-path`. WDYT?

**Comment by vmuzikar** [comment]
> The option I think is least useful is `management-relative-path`.

@ahus1 Thank you for the review! I agree. I don't have a relevant use case in mind that would benefit from changing the context root / relative path for metrics and health.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import io.sundr.builder.annotations.Buildable;
+import org.keycloak.operator.Constants;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class ManagementSpec {
+
+    @JsonPropertyDescription("If separate interface/port should be used for exposing the management endpoints.")
+    private boolean enabled = false; // TODO false for now
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java:29]: I think we could be opinionated here and not have a first class citizen option for it.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    @JsonPropertyDescription("A common root path for management endpoints.")
+    private String relativePath = "/";
+
+    @JsonPropertyDescription("Port of the management interface.")
+    private Integer port = Constants.KEYCLOAK_MANAGEMENT_PORT;
+
+    @JsonPropertyDescription("Host of the management interface.")
+    private String host;
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java:38]: Similar to above. I think we could be opinionated here and not have a first class citizen option for it.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java (hunk 2 above)
Comment (mabartos) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java:29]: Ok, so we should consider having it enabled by default, right? I think it's reasonable, but even when users try to disable it via the `additionalOptions`, the probe attributes must be changed as well. 

But I don't see a big problem here as the management interface should be the default approach for metrics and health exposure. In that case, we could direct them with some workarounds if necessary.

Comment (ahus1) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java:29]: > Ok, so we should consider having it enabled by default, right?

+1 for enabling it be default, as it would support the "use safe default" approach. When doing this, we should mention it in the upgrading guide and release notes.

**Comment by mabartos** [comment]
It required more work than previously expected. Once tests pass, I'll mark it as ready for review. Moreover, I'll summarize a little bit what changes have been proposed.

**Comment by mabartos** [comment]
Even when the test coverage is comprehended, I'd suggest testing the HealthDistTest and MetricsDistTest only on the management interface, as it should cover the suggested case.

```
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 209.912 s - in org.keycloak.it.cli.dist.HealthDistTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 207.136 s - in org.keycloak.it.cli.dist.HealthManagementDistTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 181.578 s - in org.keycloak.it.cli.dist.MetricsDistTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 181.163 s - in org.keycloak.it.cli.dist.MetricsManagementDistTest
```
You can see the cost to run them(~6.5min) is relatively high despite the value it can bring.

**Comment by mabartos** [comment]
@ahus1 @keycloak/cloud-native PR should be ready for review. Thank you!

I put the summary in the PR description.

File: quarkus/container/Dockerfile [hunk 4]
```diff
@@ -25,6 +25,9 @@ ENV LANG en_US.UTF-8
 # Flag for determining app is running in container
 ENV KC_RUN_IN_CONTAINER true
 
+# Management interface is enabled by default for container
+ENV KC_MANAGEMENT_ENABLED true
```
[the remark this answers is not in the document]
Comment (ahus1) [reply quarkus/container/Dockerfile:29]: Thank you for describing this situation. I see the point for enabling health and management port by default when using containers and the KC Operator. 

Taking a step back, I assume the health was disabled by default only because we didn't have a separate management port at that time. If we would have had it back then, we would have enabled it by default. 
I also see it as a surprise to users that it behaves differently in a container when running standalone. I remember that Wildfly also opened multiple ports by default. 

So let's rethink it, and make the management port enabled in all environment by changing the default. No need for additional "hidden" ENV variables for the container. Users can still choose to disable it if they don't want it in either setup. 

Going one step further, I see no management port enabled and exposing health + metrics on the regular port as a bad practice and it should be deprecated. It could then be removed in a future release (it is hard to imagine a setup where you would actually want to do that). Once we ensure it is never active on the regular port, we could start enabling health (and possibly metrics) by default. 

TL;DR: Enabling the management port everywhere by default is IMHO the way to go for this PR. Everything else described above is out-of-scope here.

**Review by ahus1 — COMMENTED** [review]
I've tested KC, but not the Operator. The change to KC behave OK for me. I added a comment about the management port to possibly have it enabled in all places. 

Leaving it as a "comment" only due to the ongoing discussion about the management port being enabled by default. Would support either way with a preference to have it enabled in all environments.

File: quarkus/container/Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (mabartos) [reply quarkus/container/Dockerfile:29]: > I also see it as a surprise to users that it behaves differently in a container when running standalone. I remember that Wildfly also opened multiple ports by default.

Yep, it'd be better to unify it. 

> So let's rethink it, and make the management port enabled in all environment by changing the default. No need for additional "hidden" ENV variables for the container. Users can still choose to disable it if they don't want it in either setup.

Sounds reasonable to me, thanks @ahus1!

File: quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java
Comment (vmuzikar) [inline quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: I was thinking about this more and I think we might be giving the user too many options and perhaps we could be opinionated more. A like what @ahus1 suggested:
> I see no management port enabled and exposing health + metrics on the regular port as a bad practice and it should be deprecated

What I'd suggest is:
* Do not let users to enable/disable the management port directly (also considering `--management-enabled` might be a bit ambigous for an opnion name). I'd argue we can just make and document the breaking change that health and metrics are now exposed at a different port. If we want to provide users with some deprecation period, we could introduce a specific option just for that (something like `--legacy-observability-interface=true|false`). But I don't think we strictly need that.
* I'd possibly reconsider introducing some of the "less needed" options to keep it simple for now. That concerns `management-relative-path` and `management-host`. It's always simpler to add it later (if we see the demand for it) than remove it.
* To keep it simple I would even take a step further and reconsider the HTTPS settings, I'd just inherit the setting (cert/key) from the main HTTP listener. We don't have a separate TLS setting for backend channel either (apps can be accessing KC via a backend hostname). We could, however, add some option to disable inheriting the TLS setting so that the management interface is exposed via HTTP if needed (`--management-https-enabled`).

WDYT @keycloak/cloud-native @ahus1?

Comment (mabartos) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: @vmuzikar Thanks for the review!

> Do not let users to enable/disable the management port directly (also considering --management-enabled might be a bit ambigous for an opnion name). I'd argue we can just make and document the breaking change that health and metrics are now exposed at a different port. If we want to provide users with some deprecation period, we could introduce a specific option just for that (something like --legacy-observability-interface=true|false). But I don't think we strictly need that.

@vmuzikar If we agree on having the management interface enabled by default, we do not need the "supported" option `management-enabled` as you said. It's true that we don't need to mention anywhere the removal of some property, which is not basically part of the supported options, whereas for `management-enabled` it'd be worse. 

However, I'd opt for having some backdoor for it for now, thus creating a hidden option for disabling the mgmt interface and providing some more time for proper migration.

So +1 to have such an option.

> I'd possibly reconsider introducing some of the "less needed" options to keep it simple for now. That concerns `management-relative-path` and `management-host`. It's always simpler to add it later (if we see the demand for it) than remove it.

I agree with removing `management-host` option for now, but I'd keep the `management-relative-path` as I can imagine cases where customers might use it. Either for emphasizing it's the management enpoint, or based on their API conventions. But I'm not strongly convinced :))

> To keep it simple I would even take a step further and reconsider the HTTPS settings, I'd just inherit the setting (cert/key) from the main HTTP listener. We don't have a separate TLS setting for backend channel either (apps can be accessing KC via a backend hostname).

I'm not sure whether it'd be simpler. AFAIK, in order to map the keycloak/quarkus props to the different ones, we'd still need property mappers to do so. As no mapping is done on Quarkus side for `quarkus.http-*` and `quarkus.management-*` props, we need to do it on our own. Or we would need to extend the current property mappers to support multiple properties mapping to (`to()` in the property mapper builder).

But it seems to me quite out of the scope of this PR. 
However, if the number of additional options would be the cause for worse UX, we can just hide them. @vmuzikar WDYT? 

> We could, however, add some option to disable inheriting the TLS setting so that the management interface is exposed via HTTP if needed (--management-https-enabled).

I have been thinking about it lately, and I think it might be better not to do it as we should use HTTPS whenever possible and comply with the secure-by-default "principle". Quarkus also does not provide any option to disable HTTPS for the management interface, and probably for a reason. I mean, if there's the possibility to use HTTPS, just use it. 

Or do you have some use case when it'd be suitable to use HTTP for the mgmt interface and HTTPS for the main server? WDYT?

Comment (ahus1) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: While I and 100% for making the management port enabled by default, I also see a definitive need to have a CLI option to disable the management port, as some people want to close down Keycloak as much as possible, even the management port. So this option might be called `--management-enabled` or `--management-port-enabled`. 

I don't have a strong opinion on HTTPS on the management port, and I'd say the current options seem to be a good fit what people will do with it overall. If in doubt, be in line what Quarkus does. I know that we will need TLS, and I can imagine that some people want to use client certificates on the management port as they consider this the "gold standard" for authenticating machine-to-machine. And they will want to have a separate TLS configuration as the management port is accessed directly (thereby having a different hostname) and they want to verify the hostname and the presented certificate as that is part of the regular TLS checks. The regular TLS port is behind a reverse proxy and therefore has a different certificate and hostname associated.

Comment (mabartos) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: After considering your inputs, as we will support disabling the management interface, I'd probably also keep the `--management-enabled` option there. 

I'd keep the HTTPS options, as users might easily use them for their use cases, as Alex mentioned. These options are semantically correct, and there's only a tiny probability that they're gonna be removed. The only case would be that Quarkus would drop some of them, but I kinda doubt it. 

@vmuzikar Is it ok for you to keep them?

Comment (ahus1) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: I agree that options (1) and (2) are valid, assuming that "Health and/or metrics enabled and NOT exposed on the management port" also implies that the management port is not enabled at all. Then (2) is the perfect option for those migrating from the old behavior. 

I'd strongly advocate against (3) where we would expose the admin console on the same port as the management endpoints for metrics and probes: While metrics and probes would be accessible from inside the cluster only, the admin interface would be accessible from outside of the cluster. So let's not future-proof for (3) for now.

Comment (vmuzikar) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: @ahus1 Admin console was just a bad example. :) But I do still believe there might be something else we'd want to expose on management port independently of health/metrics.

Comment (ahus1) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: ok, I can agree with that.

**Comment by mabartos** [comment]
### Proposed amendments
 
After the discussion above and internal discussion with @vmuzikar, it makes sense to me to make these changes:

1. Having an explicit toggle for the health/metrics exposure on mgmt interface
2. NOT having the general `management-enabled` toggle for the whole mgmt interface
3. If nothing is exposed on the mgmt interface, it is turned off
4. Kepp the TLS options
5. Mgmt relative path inherits the `http-relative-path` - I don't have strong objections against it.

cc: @vmuzikar @ahus1

File: quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java
Comment (mabartos) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:1]: @vmuzikar @ahus1 Thanks, guys, for the valuable discussion. Summary of amendments described below: https://github.com/keycloak/keycloak/pull/27629#issuecomment-2031903050

**Comment by vmuzikar** [comment]
@mabartos Thank you for the summary. @ahus1 Do you have any objections against 1-5?

> I'd like to follow the Quarkus approach, having it as a separate concept and not mixing it together

+1 for a separated category. That totally makes sense to me from the readability perspective. What I meant was just the prefix of the option names. To me `management` sounds a bit ambiguous in Keycloak context. I realize Quarkus uses this naming but I'd argue it doesn't mean it's the best UX for Keycloak use case as well. ;) That's why I was proposing `http-management` to make it clear it's relevant to an HTTP(S) interface.

**Comment by mabartos** [comment]
> What I meant was just the prefix of the option names. 

@vmuzikar Oh, I see. Ok, makes sense. We can mirror the convention set for Keycloak HTTP options - `http-management-*` and `https-management-*`.

**Comment by vmuzikar** [comment]
> We can mirror the convention set for Keycloak HTTP options

Exactly. :)

**Review by ahus1 — DISMISSED** [review]
Thank you for this change, happy to see this in a good state to be merged.

File: quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+ * Options for the management interface that handles management endpoints (f.e. health and metrics endpoints)
+ */
+public class ManagementOptions {
+
+    public static final Option<Boolean> LEGACY_OBSERVABILITY_INTERFACE = new OptionBuilder<>("legacy-observability-interface", Boolean.class)
+            .category(OptionCategory.MANAGEMENT)
+            .description("If metrics/health endpoints should be exposed on the main HTTP server (not recommended). If set to true, the management interface is disabled.")
+            .defaultValue(Boolean.FALSE)
+            .buildTime(true)
+            .build();
```
Comment (vmuzikar) [inline quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:32]: Should we mark this as deprecated?

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import io.sundr.builder.annotations.Buildable;
+import org.keycloak.operator.Constants;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class ManagementSpec {
+
+    @JsonPropertyDescription("A common root path for management endpoints.")
+    private String relativePath = "/";
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java:29]: I would not expose this as first class citizen. We do not expose it for [normal HTTP interface](https://github.com/keycloak/keycloak/blob/553d7c296c16f7c5fd0e995ea4743a60a66daaa6/operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/HttpSpec.java), so I'd align it with that.

File: quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java (hunk 5 above)
Comment (mabartos) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:32]: I was thinking about it and was a little bit skeptical about introducing a 'NEW DEPRECATED' option, but I don't have strong objections against it. Probably it'd make sense as it's only a temporary option.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java (hunk 6 above)
Comment (mabartos) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java:29]: ok, no problem

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/KeycloakSpec.java [hunk 7]
```diff
@@ -105,6 +106,10 @@ public class KeycloakSpec {
     @JsonPropertyDescription("In this section you can configure Keycloak's reverse proxy setting")
     private ProxySpec proxySpec;
 
+    @JsonProperty("management")
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/KeycloakSpec.java:109]: One last thing. :) Should it be like this?
```suggestion
    @JsonProperty("httpManagement")
```

File: quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java (hunk 5 above)
Comment (vmuzikar) [reply quarkus/config-api/src/main/java/org/keycloak/config/ManagementOptions.java:32]: We're deprecating exposing the metrics/health at the normal HTTP interface. This option is to turn back on this deprecated behaviour and we know we'll remove it in the future. Than IMHO qualifies it to be a deprecated new option. :)

**Comment by mabartos** [comment]
Comments should be addressed now.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java [hunk 8]
```diff
@@ -164,6 +164,8 @@ protected List<String> getArgs(Map<String, String> env) {
         if (suiteContext.get().isAuthServerMigrationEnabled()) {
             commands.add("--hostname-strict=false");
             commands.add("--hostname-strict-https=false");
+        } else { // Do not set management port for older versions of Keycloak for migration tests - available since Keycloak ~22
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java:167]:
```suggestion
        } else { // Do not set management port for older versions of Keycloak for migration tests - available since Keycloak 25
```

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ManagementHttpsDistTest.java
Comment (vmuzikar) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ManagementHttpsDistTest.java:1]: I wonder if we're missing some integration tests for the TLS options. We're relying on Quarkus in terms of functionality but we don't know if we map the correct options. We'd never know if Quarkus e.g. renamed something.

On the other hand it seems we have the same problem with the TLS options for standard HTTP interface. Hence it's not blocking from my perspective.

**Review by vmuzikar — APPROVED** [review]
@mabartos Thank you for the changes! I reviewed the rest of the code and added some comments, none of them is really blocking from my perspective.

LGTM.

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java [hunk 9]
```diff
@@ -232,6 +234,21 @@ void configureProfile(KeycloakRecorder recorder) {
...
+                                      NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem,
+                                      KeycloakRecorder recorder) {
+        final var path = Configuration.getOptionalKcValue(ManagementOptions.HTTP_MANAGEMENT_RELATIVE_PATH.getKey()).orElse("/");
+
+        routes.produce(nonApplicationRootPathBuildItem.routeBuilder()
+                .management()
+                .route(path)
+                .handler(recorder.getManagementHandler())
+                .build());
+    }
```
[the remark this answers is not in the document]
Comment (mabartos) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:250]: @vmuzikar Could you please elaborate a little bit on your concern? This step is present there due to the ability to show users they're accessing the management port right. It means when the req is sent to the management port, the HTTP 200 OK is returned with the text 'Keycloak Management Interface'. 

@ahus1 suggested it before and I like the idea as it improves the overall UX around mgmt interface.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java (hunk 8 above)
Comment (mabartos) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java:167]: I'm just wondering why I put 22 there :grin: You're right.

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (vmuzikar) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:250]: Ah, right. Missed the reasoning in the discussion above, thanks for the pointer. Makese sense to me. 👍

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ManagementHttpsDistTest.java
Comment (mabartos) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ManagementHttpsDistTest.java:1]: I understand your concern. I assume you're ok doing it in a follow-up task. I'll create some.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java (hunk 8 above)
Comment (mabartos) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java:167]: @vmuzikar Are you ok with merging it as is and amending it later? IMHO, it's not worth executing the whole CI again only due to this.

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ManagementHttpsDistTest.java
Comment (mabartos) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ManagementHttpsDistTest.java:1]: Created follow-up task: https://github.com/keycloak/keycloak/issues/28404

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java (hunk 8 above)
Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/AbstractQuarkusDeployableContainer.java:167]: Works for me. 👍
